File: 729-my-calendar-i/729-my-calendar-i.cpp
class MyCalendar {
public:
        unordered_map<int,int> mp;
    bool book(int start, int end) 
    {
			
            
        for(auto it : mp)
        {
            if(start < it.second && end > it.first)
            {
                return false;
            }
        }
        mp[start]=end;
        return true;
    }
};

/**
 * Your MyCalendar object will be instantiated and called as such:
 * MyCalendar* obj = new MyCalendar();
 * bool param_1 = obj->book(start,end);
 */

File: 729-my-calendar-i/729-my-calendar-i.java
class MyCalendar 
{
TreeSet<int[]> ts = new TreeSet<>((a, b) -> Integer.compare(a[0], b[0]));
    
    public boolean book(int start, int end) 
    {
        int[] i = new int[]{start, end - 1};
        int[] f = ts.floor(i);
        int[] c = ts.ceiling(i);
            
        if (f != null && f[1] >= start) 
                
                return false;
            
        if (c != null && c[0] <= end - 1) 
                
                return false;
       
            ts.add(i);
        return true;
}
}
/**
 * Your MyCalendar object will be instantiated and called as such:
 * MyCalendar obj = new MyCalendar();
 * boolean param_1 = obj.book(start,end);
 */